### DISCUSSION TITLE
Multiple problems and an error with option picker

### DISCUSSION THREAD

**Discussion by gerhardmpl** [Budibase/budibase#4721] [Help]
Inspired by #4418, I created three tables

`tasks` (titel: text, priority: number 0-9, estimate: number 0-9)
`priority` (id: number 0-9, name: text, name_display: formular, {{id}} - {{name}})
`estimate` (id: number 0-9, name: text, name_display: formular, {{id}} - {{name}})

My goal is to have one definition of priority and estimate and use that throughout the App based on the concept in #4418, avoiding relationships. In the Design section of  e.g. `/tasks/new/row`, I add a new Container because later I want to have the priority and estimate option picker in one row

```
/tasks/new/row
    Form
        Breadcrumps Container
        Titel Container
        Field Group
            name
            Options container (direction: column)
                Priority Data Provider (data: priority, no filter, sort on name_display, ascending)
                    Priority Picker (field priority, option source - Data provider - priority)
                Estimate Data Provider (data: estimate, no filter, sort on name_display, ascending)
                    Estimate Picker (field estimate, option source - Data provider - estimate)
```
The **first** problem occurs when adding the Priority Options Picker. It throws an `Error updating component prop` as soon as I switch `Options source` from _Schema_ to _Data Provider_. I can continue and it works despite the error message.

The **second** problem occurs when I want to choose a priority, which is defined in table `priority` from `0 - none` to `3 - high`. The Priority Picker does not show the `0 - none` entry, but starts with `1 - low`. Maybe related to #4718?

The **third** problem occurs when I change the Options Container `Direction` from _column_ to _row_. Both pickers are now in one row, but much to small and unable to display the option list properly. Changing `SIZE - Width` on the picker does not work. When I switch back to `Direction` _column_, the size of both pickers do not resize, but stay that small. Maybe it is related to the first problem, because I have to delete the Options container and start all over to get back to the usable picker size (`Direction`  _column_). 

I am using BB cloud with App version 1.0.74.

**Comment by gerhardmpl** [comment]
Attached please find the example app I use to explore `Budibase`.
[MyTasks-export-1646061523437.txt](https://github.com/Budibase/budibase/files/8154427/MyTasks-export-1646061523437.txt)

**Reply by aptkingston** [reply]
Thanks the app export @gerhardmpl - that was really helpful.

**Comment by aptkingston** [comment]
I've opened a PR to fix this: https://github.com/Budibase/budibase/pull/4750

As for point 3 - the problem here is just that you're using a `FieldGroup` component. Field groups are for quickly laying out fields in simple forms, but if you want to do custom layouts and resize components individually then you shouldn't use a field group. You can just have a form, then add some containers and fields as you see fit. You'll be able to resize and lay them out properly as long as you aren't inside a field group.

**Reply by gerhardmpl** [reply]
Thank you @aptkingston for taking the time to look into my problems and for the PR. 

Regarding point 3 - I just deleted and re-created the Options Container that is now directly connected to the 'Form'  node (moved it out of the `FieldGroup` component), but the behaviour is the same when switching the Options Containers `Direction` from _row_ to _column_. Not sure if I understood you correctly though...
